### ISSUE TITLE
Touch event detection in Internet Explorer and Edge

### ISSUE THREAD

**Issue by geoloep** [Leaflet/Leaflet#5898]
An offshoot of the discussion in #5881 .

Commit 11cbfa809576c3c6e7f3bcb8631e6cf6fc06c685 broke double tap to zoom in both Microsoft browsers. It changed a pointerType check. I do not know the history behind the check but my interpretation is that it should catch all the pointerEvents that aren't of the type 'mouse'.

After the commit the check looks like:
```javascript
e.pointerType !== 'mouse' && e.MSPOINTER_TYPE_MOUSE && e.pointerType !== e.MSPOINTER_TYPE_MOUSE
```

e.MSPOINTER_TYPE_MOUSE only exists in Internet Explorer 10: [docs](https://msdn.microsoft.com/en-us/library/jj192113(v=vs.85).aspx)

In IE11 and Edge e.MSPOINTER_TYPE_MOUSE is undefined and therefore the check never passes.

pointerType is a number in IE10 and a string in all other browsers: [docs](https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent/pointerType)

For the check to support both IE10 and IE1, Edge, Chrome etc I think it should be:
```javascript
(typeof(e.pointerType) === 'string' && e.pointerType !== 'mouse') || (typeof(e.pointerType) === 'number' && e.pointerType != e.MSPOINTER_TYPE_MOUSE)
```

First it checks the type of pointerType we're dealing with and then peforms the correct comparison.

It seems to work in IE10 (VM), IE11, Edge and Firefox. 👍 The problem is that this change breaks the double tap to zoom in Chrome once again... 😫

I theorize that touch pointerEvents in Chrome should not be 'preventDefaulted', but i'm sure that would open up yet another can of worms. 

I can't dig further at this moment, so i thought i'd summarize my finding up untill now in this issue.

**Comment by perliedman** [comment]
Thanks for your detailed thoughts, after a quick read it makes sense. Unfortunately I'm in the same situation and can't dig into this right now, but this documentation will be handy to have when we find more time.

**Comment by johnd0e** [comment]
@ghost
Could you check if the issue can fixed by #7033?
